data/basics/userBalance.go

```diff
@@ -136,6 +136,49 @@ type AccountData struct {
 	// structs; allocate a copy and modify that instead.  AccountData
 	// is expected to have copy-by-value semantics.
 	Assets map[AssetIndex]AssetHolding `codec:"asset,allocbound=-"`
+
+	AppLocalStates map[AppIndex]AppLocalState `codec:"appl,allocbound=-"`
+
+	AppParams map[AppIndex]AppParams `codec:"appp,allocbound=-"`
+}
+
+// AppLocalState stores the LocalState associated with an application. It also
+// stores a cached copy of the application's LocalStateSchema so that
+// MinBalance requirements may be computed 1. without looking up the
+// AppParams and 2. even if the application has been deleted
+type AppLocalState struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	Schema   StateSchema  `codec:"hsch"`
+	KeyValue TealKeyValue `codec:"tkv"`
+}
+
+// AppParams stores the global information associated with an application
+type AppParams struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
+
+	ApprovalProgram   string      `codec:"approv,allocbound=-"`
```

Comment:
I chose a string here because in general I like not having to worry about nil checking in functions like `Clone` below. But `[]byte` is probably more appropriate.